modules/oe_content_event/oe_content_event.module

```diff
@@ -7,76 +7,44 @@
 
 declare(strict_types = 1);
 
+use Drupal\Core\Entity\EntityInterface;
 use Drupal\Core\Form\FormStateInterface;
-use Drupal\node\Entity\Node;
+use Drupal\oe_content_event\EventUtilities;
 
 /**
  * Implements hook_entity_type_alter().
  */
 function oe_content_event_entity_type_alter(array &$entity_types) {
   // Add validation constraints to the node entity.
-  if (!empty($entity_types['node'])) {
-    $entity_types['node']->addConstraint('EventFieldsRequired');
-  }
+  $entity_types['node']->addConstraint('EventFieldsRequired');
 }
 
 /**
- * Implements hook_form_BASE_FORM_ID_alter().
+ * Implements hook_node_presave().
  */
-function oe_content_event_form_node_form_alter(&$form, FormStateInterface $form_state, $form_id) {
-  if (!in_array($form_id, ['node_oe_event_form', 'node_oe_event_edit_form'])) {
-    return;
+function oe_content_event_node_presave(EntityInterface $entity) {
+  // Make sure that we don't save both concurrent field values.
+  if ($entity->bundle() === 'oe_event') {
+    $value = $entity->get('oe_event_organiser_is_internal')->value;
+    if ($value) {
+      $entity->get('oe_event_organiser_name')->setValue(NULL);
+    }
+    elseif ($value === 0) {
+      $entity->get('oe_event_organiser_internal')->setValue(NULL);
+    }
```

Comment:
This looks strange, since we know that the value is either 1 or 0 can't we simply do the following?
```php
    if ($entity->get('oe_event_organiser_is_internal')->value === 1) {
      $entity->get('oe_event_organiser_name')->setValue(NULL);
    }
    else {
      $entity->get('oe_event_organiser_internal')->setValue(NULL);
    }
```

Reply:
This only works if the oe_event_organiser_is_internal is a required field.